### PR TITLE
use pypa github action

### PR DESCRIPTION
**Pull request by gmartinonQM**
# Description

- reorder instructions in release checklist
- use pypa publish github  action for continuous deployment

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `flake8 . --exclude=doc`
- [x] Typing passes successfully : `mypy mapie examples --strict`
- [x] Unit tests pass successfully : `pytest -vs --doctest-modules mapie`
- [x] Coverage is 100% : `pytest -vs --doctest-modules --cov-branch --cov=mapie --pyargs mapie`
- [x] Documentation builds successfully : `cd doc; make clean; make html`

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -13,14 +13,14 @@ jobs:
...
-      run: |
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
+    - name: Build package
+      run: python setup.py sdist bdist_wheel
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
+      with:
+        user: __token__
+        password: ${{ secrets.PYPI_API_TOKEN_VTA }}
```
Comment (vtaquet) [inline .github/workflows/publish.yml:26]: Le password ne prend que mon token ?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -13,14 +13,14 @@ jobs:
...
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN_VTA }}
-      run: |
-        python setup.py sdist bdist_wheel
-        twine upload dist/*
+    - name: Build package
+      run: python setup.py sdist bdist_wheel
+    - name: Publish package
+      uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
```
Comment (vtaquet) [inline .github/workflows/publish.yml:23]: A quoi correspond la suite de nombres ? Une clé sha ?

**Review by vtaquet — COMMENTED** [review]
Deux questions

File: .github/workflows/publish.yml (hunk 1 above)
Comment (gmartinonQM) [reply .github/workflows/publish.yml:26]: Oui, ça pour le coup ça n'a pas changé. PyPi exige un login et un mot de passe, donc c'est ton compte qui publie à chaque release. Ce n'est pas un souci

File: .github/workflows/publish.yml (hunk 2 above)
Comment (gmartinonQM) [reply .github/workflows/publish.yml:23]: Yes, c'est ce qui est recommandé ici (section Pro Tip) : https://github.com/pypa/gh-action-pypi-publish
C'est le sha par défaut quand tu enclenches l'action manuellement dans github

**Pull request merged**